File: src/main/java/jaba/set/ArrayBasedSet.java
package jaba.set;

import jaba.bags.ArrayBasedBag;

public class ArrayBasedSet<Item> extends ArrayBasedBag<Item> implements jaba.set.Set<Item> {

    public ArrayBasedSet(int capacity) {
        super(capacity);
    }

    @Override
    public boolean contains(Object item) {
        for (Object element : super.internalArray) {
            if (element.equals(item)) {
                return true;
            }
        }
        return false;
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public void add(Item item) {
        if (!contains(item)) {
            super.add(item);
        }
    }
}

File: src/main/java/typescript/set/ArrayBasedSet.ts
import {ArrayBasedBag} from "../bags/ArrayBasedBag";

export class ArrayBasedSet<Item> extends ArrayBasedBag<Item> {

    constructor(capacity: number) {
        super(capacity);
    }

    /**
     * {@inheritDoc}
     */
    contains(item: Item): boolean {
        let index = this.internalArray.indexOf(item);
        return index !== -1;
    }

    /**
     * {@inheritDoc}
     */
    public add(item: Item): void {
        if (!this.contains(item)) {
            super.add(item);
        }
    }
}
